### ISSUE TITLE
httpapi and non-root Zabbix WebUI location

### ISSUE THREAD

**Issue by BGmot** [ansible-collections/community.zabbix#829]
Inspired by https://github.com/ansible-collections/community.zabbix/issues/828
In all my tests with httpapi I used Zabbix server in docker and its REST API URL was always at "root" of Web server `http://localhost/api_jsonrpc.php` but during standard Zabbix Web UI installation it goes with `http://<hostname>/zabbix`, moreover anybody can install Zabbix Web UI to be served using totally custom path, like `http://<hostname>/mybelovedmonitoring`.
At this moment I don't know how to specify URL path if this collection's modules used with httpapi instead of zabbix-api. Need to do some research.

**Comment by D3DeFi** [comment]
Ouch, this is a big blocker when it comes to httpapi adoption. I though this was automatic and that it doesn't matter from which host I am running modules/roles.

I did not notice this previously - https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L48 - this is something that I wouldn't want to implement on my side probably as I am using dedicated ansible controller. 

On the other hand it shouldn't be much of a problem, but our documentation needs to reflect this more clearly (at least the README portion, EXAMPLE blocks in modules also doesn't mention this).

**Comment by BGmot** [comment]
I'll handle this "path" addition, already tested, just need to update all the docs and create a PR.

> I did not notice this previously - https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L48 - this is something that I wouldn't want to implement on my side probably as I am using dedicated ansible controller.

Sorry, what do you mean by that?

**Comment by D3DeFi** [comment]
> I'll handle this "path" addition, already tested, just need to update all the docs and create a PR.
> 
> > I did not notice this previously - https://github.com/ansible-collections/community.zabbix/blob/main/roles/zabbix_agent/tasks/api.yml#L48 - this is something that I wouldn't want to implement on my side probably as I am using dedicated ansible controller.
> 
> Sorry, what do you mean by that?

I understand that it is now required to execute zabbix modules that are using httpapi from Zabbix server due to:
```yaml
    ansible_network_os: community.zabbix.zabbix
    ansible_connection: httpapi
```

Am I correct? I would assume that this with combination of `delegate_to` means that Ansible connects via SSH to Zabbix server and runs module from there, which then creates HTTP request on localhost, correct or not? 

If above is true, what if I need this HTTP request to be made from my Ansible controller and not from the Zabbx server directly?

**Comment by BGmot** [comment]
No, this is interesting case. Httpapi treats `delegate_to` as a hostname to connect to via HTTP(S), the code is executed on Ansible controller.

**Comment by D3DeFi** [comment]
> No, this is interesting case. Httpapi treats `delegate_to` as a hostname to connect to via HTTP(S), the code is executed on Ansible controller.
> 
> Maybe the reasoning behind this is define all your Zabbix servers in an Inventory as regular hosts.

Oh, ok got it. I rest my case! 😅
